totalRP3/modules/register/characters/PlayerMapScan/PlayerMapPin.lua

```diff
@@ -0,0 +1,121 @@
+-- Create the pin template, above group members
+TRP3_PlayerMapPinMixin = AddOn_TotalRP3.MapPoiMixins.createPinTemplate(
+	AddOn_TotalRP3.MapPoiMixins.GroupedCoalescedMapPinMixin, -- Use coalesced grouped tooltips (show multiple player names)
+	AddOn_TotalRP3.MapPoiMixins.AnimatedPinMixin -- Use animated icons (bounce in)
+);
+
+-- Expose template name, so the scan can use it for the MapDataProvider
+TRP3_PlayerMapPinMixin.TEMPLATE_NAME = "TRP3_PlayerMapPinTemplate";
+
+--- This is called when the data provider acquire a pin, to transform poiInfo received from the scan
+--- into display info to be used to decorate the pin.
+function TRP3_PlayerMapPinMixin:GetDisplayDataFromPoiInfo(poiInfo)
+	local characterID = poiInfo.sender;
+	local displayData = {};
+	local isSelf = characterID == TRP3_API.globals.player_id;
+
+	if not isSelf and (not TRP3_API.register.isUnitIDKnown(characterID) or not TRP3_API.register.hasProfile(characterID)) then
+		-- Only remove the server name from the sender ID
+		displayData.playerName = characterID:gsub("%-.*$", "");
+		return displayData;
+	end
+
+	local profile;
+	if isSelf then
+		-- Special case when seeing ourselves on the map (DEBUG)
+		profile = TRP3_API.profile.getPlayerCurrentProfile().player;
+		displayData.iconAtlas = "PlayerPartyBlip";
+		displayData.iconColor = Ellyb.ColorManager.CYAN;
+		displayData.categoryName = loc.REG_RELATION .. ": " .. Ellyb.ColorManager.CYAN("SELF");
+		displayData.categoryPriority = -huge;
+	else
+		profile = TRP3_API.register.getUnitIDCurrentProfile(characterID);
+
+		--region Player relationship
+		local relation = TRP3_API.register.relation.getRelation(TRP3_API.register.getUnitIDProfileID(characterID));
+		if relation ~= TRP3_API.register.relation.NONE then
+			local relationShipColor = TRP3_API.register.relation.getColor(relation);
+			-- Swap out the atlas for this marker.
+			displayData.iconAtlas = "PlayerPartyBlip";
+			displayData.iconColor = relationShipColor;
+
+			-- Store the relationship on the marker itself as the category.
+			displayData.categoryName = loc.REG_RELATION .. ": " .. relationShipColor(loc:GetText("REG_RELATION_".. relation));
+			displayData.categoryPriority = TRP3_API.globals.RELATIONS_ORDER[relation] or -huge;
+		end
+		--endregion
+	end
+
+	--region Player name
+	displayData.playerName = TRP3_API.register.getCompleteName(profile.characteristics, characterID, true);
+
+	if profile.characteristics then
+		if profile.characteristics.CH then
+			local color = Ellyb.Color.CreateFromHexa(profile.characteristics.CH);
+			displayData.playerName = color(displayData.playerName);
+		end
+		if profile.characteristics.IC then
+			displayData.playerName = TRP3_API.utils.str.icon(profile.characteristics.IC, 15) .. " " .. displayData.playerName;
+		end
+	end
+	--endregion
+
+	return displayData
+end
+
+--- This is called by the data provider to decorate the pin after the base pin mixin has done its job.
+---@param poiInfo {position:Vector2DMixin, sender:string}
```

Comment:
Parameter is displayData, not poiInfo, the content of it is probably incorrect as well.

Reply:
That param is actually the one from the previous function `GetDisplayDataFromPoiInfo`, I would guess you separated this part of the function later and didn't move the doc part.